sql/src/main/java/io/crate/analyze/relations/RelationNormalizer.java

```diff
@@ -119,6 +119,14 @@ public AnalyzedRelation visitMultiSourceSelect(MultiSourceSelect multiSourceSele
     public AnalyzedRelation visitTwoRelationsUnion(TwoRelationsUnion twoTableUnion, Context context) {
         process(twoTableUnion.first(), context);
         process(twoTableUnion.second(), context);
+
+        // PushDown  limit
+        Optional<Symbol> pushDownLimit = Limits.add(twoTableUnion.querySpec().limit(),
```

Comment:
can you extend the comment to explain more on a semantic level what is happening here? why limit + offset?